src/umpire/Umpire.cpp

```diff
@@ -98,6 +113,141 @@ bool pointer_contains(void* left_ptr, void* right_ptr)
   }
 }
 
+bool is_host_pageable()
+{
+#if defined(UMPIRE_ENABLE_HIP)
+  hipDeviceProp_t props;
+  int hdev = 0;
+  hipGetDevice(&hdev);
+
+  //Check whether HIP can map host memory.
+  hipGetDeviceProperties(&props, hdev);
```

Comment:
Can these device-implementation-specific calls be placed in to static methods of the specific resource instead of being placed here?

Reply:
They cannot be static since some info (e.g. device number) would be a property of a specific resource instance.

Reply:
Good idea, stay tuned! :)